src/tests/ftest/control/dmg_pool_query_test.yaml

```diff
@@ -14,13 +14,13 @@ server_config:
     scm_class: dcpm
     scm_list: ["/dev/pmem0"]
   transport_config:
-    allow_insecure: True
+    allow_insecure: False
 agent_config:
   transport_config:
-    allow_insecure: True
+    allow_insecure: False
 dmg:
   transport_config:
-    allow_insecure: True
+    allow_insecure: False
```

Comment:
Since we know this test passes with `allow_insecure: False` (https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-6015/1/artifact/Functional%20Hardware%20Small/control/dmg_pool_query_test.py/) and with `allow_insecure: True` (https://build.hpdd.intel.com/job/daos-stack/job/daos/job/master/4840/artifact/Functional%20Hardware%20Small/control/dmg_pool_query_test.py/), we should enable launch.py to control the setting fully with its `-ins` command line option. This means removing the current restriction of always running this test in secure mode and deleting lines 16-23.